python/ray/_private/usage/usage_lib.py

```diff
@@ -182,6 +193,106 @@ def put_cluster_metadata(gcs_client, num_retries) -> None:
     return metadata
 
 
+def get_cluster_status(gcs_client, num_retries) -> dict:
+    """Get the current status of this cluster."""
+    cluster_status = ray._private.utils.internal_kv_get_with_retry(
+        gcs_client,
+        ray.ray_constants.DEBUG_AUTOSCALING_STATUS,
+        namespace=None,
+        num_retries=num_retries,
+    )
+    if not cluster_status:
+        return {}
+
+    result = {}
+    to_GiB = 1 / 2 ** 30
+    cluster_status = json.loads(cluster_status.decode("utf-8"))
+    if (
+        "load_metrics_report" not in cluster_status
+        or "usage" not in cluster_status["load_metrics_report"]
+    ):
+        return {}
+
+    usage = cluster_status["load_metrics_report"]["usage"]
+    if "CPU" in usage:
+        result["num_cpus"] = usage["CPU"][1]
+    if "GPU" in usage:
+        result["num_gpus"] = usage["GPU"][1]
+    if "memory" in usage:
+        result["total_memory_gb"] = usage["memory"][1] * to_GiB
+    if "object_store_memory" in usage:
+        result["total_object_store_memory_gb"] = (
+            usage["object_store_memory"][1] * to_GiB
+        )
+    return result
+
+
+def get_cluster_config(cluster_config_file_path) -> dict:
+    """Get the static cluster (autoscaler) config used to launch this cluster."""
+
+    def get_instance_type(node_config):
+        if not node_config:
+            return None
+        if "InstanceType" in node_config:
+            # aws
+            return node_config["InstanceType"]
+        if "machineType" in node_config:
+            # gcp
+            return node_config["machineType"]
+        if (
+            "azure_arm_parameters" in node_config
+            and "vmSize" in node_config["azure_arm_parameters"]
+        ):
```

Comment:
can we make this support k8s?

Reply:
@wuisawesome Oh so k8s cluster also has `~/ray_bootstrap_config.yaml` file? Could you give me some pointers to what the content is for k8s cluster so I can know how to parse it.

Reply:
In kuberay the provider type is "kuberay": https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kuberay/autoscaling_config.py#L99

Reply:
KubeRay autoscaling support no longer stores configs on disk, see https://github.com/ray-project/ray/pull/22348

Reply:
If you want to check if we're in a K8s pod, one option is to check for existence of the env `KUBERNETES_SERVICE_HOST` 
https://github.com/ray-project/ray/blob/cc1728120f7d49b0016d190971bc8056d3245c5d/dashboard/modules/reporter/reporter_agent.py#L33